### PR TITLE
[Playground] Datastore namespace, Helm deployment, Ingress timeout, REDIS type fixes

### LINKED ISSUE #24814: [Task]: Adapt Playground infrastructure to support Datastore multitenancy
### What needs to happen?

Change scripts to support custom environment variable DATASTORE_NAMESPACE for containers

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [X] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by MakarkinSAkvelon**
resolves #24814
PR request will fix followig things:
Datastore namespace - allow to change Datastore namespace from command line
Helm deployment - fix resourse limit
Ingress timeout - fix SCIO examples upload issue
REDIS type - change from Standard to Basic

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by olehborysevych** [comment]
LGTM

**Review by olehborysevych — APPROVED** [review]

**Review by rshamunov — APPROVED** [review]
LGTM

**Comment by MakarkinSAkvelon** [comment]
R @damondouglas

File: playground/terraform/build.gradle.kts [hunk 1]
```diff
@@ -392,13 +392,17 @@ tasks.register("takeConfig") {
    var ipaddrname = ""
    var d_tag = ""
    var dns_name = ""
+   var dstore_name = ""
```
Comment (damondouglas) [inline playground/terraform/build.gradle.kts:395]: Should we spell out the full `datastore_name`?

File: playground/terraform/infrastructure/variables.tf [hunk 2]
```diff
@@ -94,12 +94,12 @@ variable "redis_name" {
 
 variable "redis_tier" {
   description = "Tier of Redis"
-  default     = "STANDARD_HA"
```
Comment (damondouglas) [inline playground/terraform/infrastructure/variables.tf:97]: Why are we downgrading?

**Review by damondouglas — CHANGES_REQUESTED** [review]

File: playground/terraform/infrastructure/variables.tf (hunk 2 above)
Comment (MakarkinSAkvelon) [reply playground/terraform/infrastructure/variables.tf:97]: Hello, @damondouglas. Even on current Production we're using BASIC Redis version and it's enough

Comment (rshamunov) [reply playground/terraform/infrastructure/variables.tf:97]: We don't downgrade. Currently we use Basic Redis instance. According to monitoring, we don't utilize it enough to switch to Standard instance.

File: playground/terraform/build.gradle.kts (hunk 1 above)
Comment (MakarkinSAkvelon) [reply playground/terraform/build.gradle.kts]: Variable name was fixed

File: playground/terraform/infrastructure/variables.tf (hunk 2 above)
Comment (damondouglas) [reply playground/terraform/infrastructure/variables.tf:97]: @rshamunov What if the usage grows? What is the plan to monitor and respond to capacity issues?

**Review by damondouglas — CHANGES_REQUESTED** [review]

File: playground/terraform/infrastructure/variables.tf (hunk 2 above)
Comment (rshamunov) [reply playground/terraform/infrastructure/variables.tf:97]: @damondouglas, currently Playground use Redis for tasks that don't utilize even Basic tier capacity. We are not at risk of performance issues, even if usage of Playground environment grows significantly. Besides, we can reconfigure Playground to use new Redis instance if we need.

**Review by damondouglas — CHANGES_REQUESTED** [review]
Tried to test these changes but blocked on the following error when following instructions written https://github.com/akvelon/beam/blob/f5458ea843a53eda7d01fd43c40ddceb8d946c0d/playground/terraform/README.md

<details>
<summary>> Task :playground:terraform:terraformApplyInf FAILED</summary>


```
Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

> Task :playground:terraform:terraformApplyInf FAILED
╷
│ Warning: Resource targeting is in effect
╷
│
│ Error: Plugin did not respond
│ You are creating a plan with the -target option, which means that the
│
│ result of this plan may not represent all of the changes requested by the
│   with provider["registry.terraform.io/hashicorp/google-beta"],
│ current configuration.
│   on provider.tf line 42, in provider "google-beta":
│
│   42: provider "google-beta" {
│ The -target option is not for routine use, and is provided only for
│
│ exceptional situations such as recovering from errors or mistakes, or when
│ The plugin encountered an error, and failed to respond to the
│ Terraform specifically suggests to use it as part of an error message.
│ plugin.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain
╵
│ more details.

╵
Stack trace from the terraform-provider-google-beta_v4.53.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x5ecc01]

goroutine 40 [running]:
github.com/hashicorp/terraform-plugin-log/internal/logging.GetProviderRootLogger(...)
        github.com/hashicorp/terraform-plugin-log@v0.7.0/internal/logging/provider.go:12
github.com/hashicorp/terraform-plugin-log/tflog.Info({0x0, 0x0}, {0x3382975, 0x85}, {0x0, 0x0, 0x0})
        github.com/hashicorp/terraform-plugin-log@v0.7.0/tflog/provider.go:74 +0x61
github.com/hashicorp/terraform-provider-google-beta/google-beta.getCurrUserEmail(0xc000297800, {0xc00032c1b0?, 0x3860c60?}, 0xc000696e50)
        github.com/hashicorp/terraform-provider-google-beta/google-beta/framework_utils.go:44 +0x25f
github.com/hashicorp/terraform-provider-google-beta/google-beta.(*frameworkProvider).logGoogleIdentities(_, {_, _}, {{0x0, {0x0, 0x0}}, {0x0, {0x0, 0x0}}, {0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-provider-google-beta/google-beta/framework_config.go:1332 +0x698
github.com/hashicorp/terraform-provider-google-beta/google-beta.(*frameworkProvider).SetupClient(_, {_, _}, {{0x0, {0x0, 0x0}}, {0x0, {0x0, 0x0}}, {0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-provider-google-beta/google-beta/framework_config.go:1262 +0x28f
github.com/hashicorp/terraform-provider-google-beta/google-beta.(*frameworkProvider).ConfigureWithData(_, {_, _}, {{0x0, {0x0, 0x0}}, {0x0, {0x0, 0x0}}, {0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-provider-google-beta/google-beta/framework_config.go:47 +0x23e
github.com/hashicorp/terraform-provider-google-beta/google-beta.(*frameworkProvider).Configure(0xc000297800, {0x3869b98, 0xc000d31500}, {{0xc00069a158, 0x5}, {{{0x3870838, 0xc000fbc4b0}, {0x2da5a40, 0xc000f97b30}}, {0x3871cf8, ...}}}, ...)
        github.com/hashicorp/terraform-provider-google-beta/google-beta/framework_provider.go:918 +0x225
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ConfigureProvider(0xc000224160, {0x3869b98, 0xc000d31500}, 0xc000696ec0, 0xc000696e40)
        github.com/hashicorp/terraform-plugin-framework@v1.1.1/internal/fwserver/server_configureprovider.go:15 +0x10f
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ConfigureProvider(0xc000224160, {0x3869b98?, 0xc000d313b0?}, 0x3?)
        github.com/hashicorp/terraform-plugin-framework@v1.1.1/internal/proto5server/server_configureprovider.go:36 +0x292
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ConfigureProvider({0xc000881950, 0xc0008819b0, {0xc00014d840, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-plugin-mux@v0.8.0/tf5muxserver/mux_server_ConfigureProvider.go:25 +0x1b3
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).Configure(0xc000ed05a0, {0x3869b98?, 0xc000d30510?}, 0xc000696b00)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:556 +0x2ce
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler({0x319b2c0?, 0xc000ed05a0}, {0x3869b98, 0xc000d30510}, 0xc000bd6000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:331 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000005a0, {0x3870ac0, 0xc00074d380}, 0xc000b68000, 0xc000d289f0, 0x4c0c918, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc0000005a0, {0x3870ac0, 0xc00074d380}, 0xc000b68000, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-google-beta_v4.53.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.


FAILURE: Build failed with an exception.
```
</details>

# Environment Details
- Debian 11
- Terraform v1.3.8

File: playground/terraform/README.md [hunk 3]
```diff
@@ -116,9 +116,9 @@ gcloud container clusters get-credentials --region `chosen_location` `gke_name`
 ```
 Start the following command from the top level repository folder ("beam") to deploy the Payground infrastructure:
 ```
-./gradlew playground:terraform:gkebackend -Pproject_environment="environment_name" -Pdocker-tag="tag" -Pdns-name="playground.zone" -Psdk-tag=2.43.0
+./gradlew playground:terraform:gkebackend -Pproject_environment="environment_name" -Pdocker-tag="tag" -Pdns-name="playground.zone" -Psdk-tag=2.43.0 -Pdatastore-namespace="dev"
 ```
-Where tag - image tag for backend, playground.zone - chosen DNS for Playground, Psdk-tag - current BEAM version
+Where tag - image tag for backend, playground.zone - chosen DNS for Playground, Psdk-tag - current BEAM version, Pdstore-namespace - namespace for Datastore
```
Comment (rshamunov) [inline playground/terraform/README.md:121]: -Pdatastore_namespace ?

**Comment by MakarkinSAkvelon** [comment]
Hello, @damondouglas. The only reason why we could get such kind of an issue is just to stop the script during execution that  *.tfstate file didn't contain all the required variables (path to *.tfstate file - "beam/playground/terraform/.terraform") 
Could you try the following:
1. Remove "beam/playground/terraform/.terraform" folder
2. Ensure that "beam/playground/terraform/$environment/terraform.tfvars" contain variables according to Readme.md from the PR
3. Start the deployment script

Feel free to ping me in any chats (slack or google chat) in case of any issues

**Review by damondouglas — CHANGES_REQUESTED** [review]
I tried running through the end to end deployment and it failed:

### Error

See details below.

<details>

```
#12 [build 6/7] RUN cd /app/playground_components &&    flutter pub get -v
#12 0.298    Woah! You appear to be trying to run flutter as root.
#12 0.298    We strongly recommend running the flutter tool without superuser privileges.
#12 0.298   /
#12 0.298 📎
#12 0.302 fatal: detected dubious ownership in repository at '/opt/flutter'
#12 0.302 To add an exception for this directory, call:
#12 0.302
#12 0.302       git config --global --add safe.directory /opt/flutter
#12 0.727 [ +164 ms] executing: uname -m
#12 0.748 [  +27 ms] Exit code 0 from: uname -m
#12 0.748 [        ] x86_64
#12 0.754 [   +5 ms] executing: [/opt/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
#12 0.760 [   +6 ms] Exit code 128 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
#12 0.762 [        ] fatal: detected dubious ownership in repository at '/opt/flutter'
#12 0.762            To add an exception for this directory, call:
#12 0.762
#12 0.762               git config --global --add safe.directory /opt/flutter
#12 0.762 [   +2 ms] executing: [/opt/flutter/] git tag --points-at
#12 0.768 [   +5 ms] Exit code 128 from: git tag --points-at
#12 0.768 [        ] fatal: detected dubious ownership in repository at '/opt/flutter'
#12 0.768            To add an exception for this directory, call:
#12 0.768
#12 0.768               git config --global --add safe.directory /opt/flutter
#12 0.770 [   +1 ms] executing: [/opt/flutter/] git describe --match *.*.* --long --tags
#12 0.775 [   +5 ms] Exit code 128 from: git describe --match *.*.* --long --tags
#12 0.776 [        ] fatal: detected dubious ownership in repository at '/opt/flutter'
#12 0.776            To add an exception for this directory, call:
#12 0.776
#12 0.776               git config --global --add safe.directory /opt/flutter
#12 0.778 [   +2 ms] Could not interpret results of "git describe":
#12 0.830 [  +52 ms] Unable to locate an Android SDK.
#12 0.835 [   +4 ms] executing: [/opt/flutter/] git rev-parse --abbrev-ref HEAD
#12 0.840 [   +5 ms] Exit code 128 from: git rev-parse --abbrev-ref HEAD
#12 0.840 [        ] fatal: detected dubious ownership in repository at '/opt/flutter'
#12 0.840            To add an exception for this directory, call:
#12 0.840 
#12 0.840               git config --global --add safe.directory /opt/flutter
#12 0.884 [  +43 ms]   ╔════════════════════════════════════════════════════════════════════════════╗
#12 0.884              ║                 Welcome to Flutter! - https://flutter.dev                  ║
#12 0.884              ║                                                                            ║
#12 0.884              ║ The Flutter tool uses Google Analytics to anonymously report feature usage ║
#12 0.884              ║ statistics and basic crash reports. This data is used to help improve      ║
#12 0.884              ║ Flutter tools over time.                                                   ║
#12 0.884              ║                                                                            ║
#12 0.884              ║ Flutter tool analytics are not sent on the very first run. To disable      ║
#12 0.884              ║ reporting, type 'flutter config --no-analytics'. To display the current    ║
#12 0.884              ║ setting, type 'flutter config'. If you opt out of analytics, an opt-out    ║
#12 0.884              ║ event will be sent, and then no further information will be sent by the    ║
#12 0.884              ║ Flutter tool.                                                              ║
#12 0.884              ║                                                                            ║
#12 0.884              ║ By downloading the Flutter SDK, you agree to the Google Terms of Service.  ║
#12 0.884              ║ Note: The Google Privacy Policy describes how data is handled in this      ║
#12 0.884              ║ service.                                                                   ║
#12 0.884              ║                                                                            ║
#12 0.884              ║ Moreover, Flutter includes the Dart SDK, which may send usage metrics and  ║
#12 0.884              ║ crash reports to Google.                                                   ║
#12 0.884              ║                                                                            ║
#12 0.884              ║ Read about data we send with crash reports:                                ║
#12 0.884              ║ https://flutter.dev/docs/reference/crash-reporting                         ║
#12 0.884              ║                                                                            ║
#12 0.884              ║ See Google's privacy policy:                                               ║
#12 0.884              ║ https://policies.google.com/privacy                                        ║
#12 0.884              ╚════════════════════════════════════════════════════════════════════════════╝
#12 0.884
#12 0.899 [  +15 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
#12 0.899 [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
#12 0.899 [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
#12 0.899 [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
#12 0.902 [   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
#12 0.902 [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
#12 0.902 [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
#12 0.902 [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
#12 0.902 [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
#12 0.902 [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
#12 0.903 [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
#12 0.946 [  +43 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
#12 0.947 [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
#12 0.947 [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
#12 0.947 [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
#12 0.947 [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
#12 0.947 [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
#12 0.947 [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
#12 0.947 [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
#12 0.948 [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
#12 0.948 [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
#12 0.948 [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
#12 0.948 [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
#12 0.948 [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
#12 0.948 [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
#12 0.948 [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
#12 0.948 [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
#12 0.949 [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
#12 0.949 [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
#12 0.949 [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
#12 0.949 [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
#12 0.949 [        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
#12 1.018 [  +68 ms] Running "flutter pub get" in playground_components...
#12 1.021 [   +2 ms] Using /opt/flutter/.pub-cache for the pub cache.
#12 1.021 [        ] executing: [/app/playground_components/] /opt/flutter/bin/cache/dart-sdk/bin/dart __deprecated_pub --trace --verbose get --no-precompile
#12 1.105 [  +82 ms] FINE: Pub 2.18.1
#12 1.170 [  +66 ms] MSG : Resolving dependencies...
#12 1.192 [  +21 ms] SLVR: fact: playground_components is 0.0.1
#12 1.199 [   +6 ms] SLVR: derived: playground_components
#12 1.223 [  +23 ms] SLVR: fact: playground_components requires Flutter SDK version >=3.3.2
#12 1.228 [   +4 ms] SLVR: conflict: playground_components requires Flutter SDK version >=3.3.2
#12 1.232 [   +4 ms] SLVR: Version solving took 0:00:00.044091 seconds.
#12 1.233 [        ]     | Tried 1 solutions.
#12 1.233 [        ] FINE: Resolving dependencies finished (0.067s).
#12 1.245 [  +11 ms] ERR : The current Flutter SDK version is 0.0.0-unknown.
#12 1.245 [        ]     |
#12 1.246 [        ]     | Because playground_components requires Flutter SDK version >=3.3.2, version solving failed.
#12 1.246 [        ] FINE: Exception type: SolveFailure
#12 1.269 [  +23 ms] FINE: package:pub/src/solver/version_solver.dart 311:5               VersionSolver._resolveConflict
#12 1.270 [        ]     | package:pub/src/solver/version_solver.dart 132:27              VersionSolver._propagate
#12 1.270 [        ]     | package:pub/src/solver/version_solver.dart 96:11               VersionSolver.solve.<fn>
#12 1.270 [        ]     | ===== asynchronous gap ===========================
#12 1.270 [        ]     | dart:async                                                     Future.catchError
#12 1.270 [        ]     | package:pub/src/utils.dart 109:52                              captureErrors.wrappedCallback
#12 1.271 [        ]     | package:stack_trace                                            Chain.capture
#12 1.271 [        ]     | package:pub/src/utils.dart 122:11                              captureErrors
#12 1.271 [        ]     | package:pub/src/command.dart 189:13                            PubCommand.run
#12 1.271 [        ]     | package:args/command_runner.dart 209:27                        CommandRunner.runCommand
#12 1.272 [        ]     | package:pub/src/command_runner.dart 176:24                     PubCommandRunner.runCommand
#12 1.272 [        ]     | package:pub/src/command_runner.dart 161:20                     PubCommandRunner.run
#12 1.273 [        ]     | package:dartdev/dartdev.dart 45:56                             runDartdev
#12 1.273 [        ]     | /b/s/w/ir/cache/builder/sdk/pkg/dartdev/bin/dartdev.dart 11:9  main
#12 1.273 [        ] ---- Log transcript ----
#12 1.273 [        ] FINE: Pub 2.18.1
#12 1.273 [        ] MSG : Resolving dependencies...
#12 1.274 [        ] SLVR: fact: playground_components is 0.0.1
#12 1.274 [        ] SLVR: derived: playground_components
#12 1.275 [        ] SLVR: fact: playground_components requires Flutter SDK version >=3.3.2
#12 1.275 [   +1 ms] SLVR: conflict: playground_components requires Flutter SDK version >=3.3.2
#12 1.276 [        ] SLVR: Version solving took 0:00:00.044091 seconds.
#12 1.276 [        ]     | Tried 1 solutions.
#12 1.276 [        ] FINE: Resolving dependencies finished (0.067s).
#12 1.277 [        ] ERR : The current Flutter SDK version is 0.0.0-unknown.
#12 1.277 [        ]     |
#12 1.277 [        ]     | Because playground_components requires Flutter SDK version >=3.3.2, version solving failed.
#12 1.277 [        ] FINE: Exception type: SolveFailure
#12 1.278 [        ] FINE: package:pub/src/solver/version_solver.dart 311:5               VersionSolver._resolveConflict
#12 1.278 [        ]     | package:pub/src/solver/version_solver.dart 132:27              VersionSolver._propagate
#12 1.278 [        ]     | package:pub/src/solver/version_solver.dart 96:11               VersionSolver.solve.<fn>
#12 1.278 [        ]     | ===== asynchronous gap ===========================
#12 1.278 [        ]     | dart:async                                                     Future.catchError
#12 1.279 [        ]     | package:pub/src/utils.dart 109:52                              captureErrors.wrappedCallback
#12 1.279 [        ]     | package:stack_trace                                            Chain.capture
#12 1.279 [        ]     | package:pub/src/utils.dart 122:11                              captureErrors
#12 1.279 [        ]     | package:pub/src/command.dart 189:13                            PubCommand.run
#12 1.279 [        ]     | package:args/command_runner.dart 209:27                        CommandRunner.runCommand
#12 1.279 [        ]     | package:pub/src/command_runner.dart 176:24                     PubCommandRunner.runCommand
#12 1.280 [        ]     | package:pub/src/command_runner.dart 161:20                     PubCommandRunner.run
#12 1.280 [        ]     | package:dartdev/dartdev.dart 45:56                             runDartdev
#12 1.280 [        ]     | /b/s/w/ir/cache/builder/sdk/pkg/dartdev/bin/dartdev.dart 11:9  main
#12 1.280 [        ] ---- End log transcript ----
#12 1.283 [   +2 ms] IO  : Writing 4553 characters to text file /opt/flutter/.pub-cache/log/pub_log.txt.
#12 1.285 [   +2 ms] MSG : Logs written to /opt/flutter/.pub-cache/log/pub_log.txt.
#12 1.337 [  +51 ms] Running "flutter pub get" in playground_components... (completed in 310ms)
#12 1.338 [   +1 ms] "flutter get" took 467ms.
#12 1.341 [   +3 ms] pub get failed (1; MSG : Logs written to /opt/flutter/.pub-cache/log/pub_log.txt.)
#12 1.342 [        ]
#12 1.342            #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
#12 1.342            #1      _DefaultPub.batch (package:flutter_tools/src/dart/pub.dart:375:7)
#12 1.342            <asynchronous suspension>
#12 1.342            #2      _DefaultPub.get (package:flutter_tools/src/dart/pub.dart:256:7)
#12 1.342            <asynchronous suspension>
#12 1.342            #3      PackagesGetCommand._runPubGet (package:flutter_tools/src/commands/packages.dart:139:7)
#12 1.342            <asynchronous suspension>
#12 1.342            #4      PackagesGetCommand.runCommand (package:flutter_tools/src/commands/packages.dart:174:5)
#12 1.342            <asynchronous suspension>
#12 1.342            #5      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1209:27)
#12 1.342            <asynchronous suspension>
#12 1.342            #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
#12 1.342            <asynchronous suspension>
#12 1.342            #7      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
#12 1.342            <asynchronous suspension>
#12 1.342            #8      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:281:9)
#12 1.342            <asynchronous suspension>
#12 1.342            #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
#12 1.342            <asynchronous suspension>
#12 1.342            #10     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
#12 1.342            <asynchronous suspension>
#12 1.342            #11     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
#12 1.342            <asynchronous suspension>
#12 1.342            #12     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
#12 1.342            <asynchronous suspension>
#12 1.342            #13     main (package:flutter_tools/executable.dart:91:3)
#12 1.342            <asynchronous suspension>
#12 1.342
#12 1.342
#12 1.343 [   +1 ms] ensureAnalyticsSent: 0ms
#12 1.344 [        ] Running shutdown hooks
#12 1.345 [        ] Shutdown hooks complete
#12 1.346 [   +1 ms] exiting with code 1
#12 ERROR: process "/bin/sh -c cd /app/playground_components &&    flutter pub get -v" did not complete successfully: exit code: 1
------
 > [build 6/7] RUN cd /app/playground_components &&    flutter pub get -v:
#12 1.342            #12     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
#12 1.342            <asynchronous suspension>
#12 1.342            #13     main (package:flutter_tools/executable.dart:91:3)
#12 1.342            <asynchronous suspension>
#12 1.342
#12 1.342
#12 1.343 [   +1 ms] ensureAnalyticsSent: 0ms
#12 1.344 [        ] Running shutdown hooks
#12 1.345 [        ] Shutdown hooks complete
#12 1.346 [   +1 ms] exiting with code 1
------
Dockerfile:43
--------------------
  42 |
  43 | >>> RUN cd /app/playground_components &&\
  44 | >>>     flutter pub get -v
  45 |
--------------------
ERROR: failed to solve: process "/bin/sh -c cd /app/playground_components &&    flutter pub get -v" did not complete successfully: exit code: 1

> Task :playground:frontend:docker FAILED

FAILURE: Build failed with an exception.
```
</details>

### Environment

See environment details below:

<details>

```
====== GO ======

go version go1.20 linux/amd64

==================
====== JAVA ======

openjdk version "11.0.18" 2023-01-17
OpenJDK Runtime Environment (build 11.0.18+10-post-Debian-1deb11u1)
OpenJDK 64-Bit Server VM (build 11.0.18+10-post-Debian-1deb11u1, mixed mode, sharing)

==================

====== kubectl ======
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.26.1
Kustomize Version: v4.5.7
Server Version: v1.24.9-gke.3200
WARNING: version difference between client (1.26) and server (1.24) exceeds the supported minor version skew of +/-1

==================

====== helm ======
version.BuildInfo{Version:"v3.11.1", GitCommit:"293b50c65d4d56187cd4e2f390f0ada46b4c4737", GitTreeState:"clean", GoVersion:"go1.18.10"}

==================

====== docker ======
Client: Docker Engine - Community
 Version:           23.0.1
 API version:       1.42
 Go version:        go1.19.5
 Git commit:        a5ee5b1
 Built:             Thu Feb  9 19:46:54 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          23.0.1
  API version:      1.42 (minimum version 1.12)
  Go version:       go1.19.5
  Git commit:       bc3805a
  Built:            Thu Feb  9 19:46:54 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.16
  GitCommit:        31aa4358a36870b21a992d3ad2bef29e1d693bec
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

==================

====== terraform ======
Terraform v1.3.8
on linux_amd64

Your version of Terraform is out of date! The latest version
is 1.3.9. You can update by downloading from https://www.terraform.io/downloads.html

==================

====== gcloud ======
Google Cloud SDK 416.0.0
alpha 2023.01.30
beta 2023.01.30
bq 2.0.84
bundled-python3-unix 3.9.16
core 2023.01.30
gcloud-crc32c 1.0.0
gke-gcloud-auth-plugin 0.4.0
gsutil 5.19

==================
====== SBT ======

sbt version in this project: 1.8.2
sbt script version: 1.8.2
```
</details>

**Comment by MakarkinSAkvelon** [comment]
It's a known issue with was fixed before (but after the current PR was raised) so now rebase was completed, please deploy it one more time

File: playground/infrastructure/helm-playground/templates/autoscaling-go.yaml [hunk 4]
```diff
@@ -12,15 +12,27 @@
...
 # limitations under the License.
-apiVersion: autoscaling/v1
+apiVersion: autoscaling/v2
 kind: HorizontalPodAutoscaler
 metadata:
   name: playground-go
 spec:
-  maxReplicas: 10
-  minReplicas: 1
+  maxReplicas: 4
```
Comment (volatilemolotov) [inline playground/infrastructure/helm-playground/templates/autoscaling-go.yaml:20]: This should be configurable and default should be provided via value file

Comment (MakarkinSAkvelon) [reply playground/infrastructure/helm-playground/templates/autoscaling-go.yaml:20]: Helm variables were created

**Comment by rshamunov** [comment]
Replaced by #25994

**Pull request closed without merging**